app/assets/javascripts/discourse/app/components/user-menu/notification-item.hbs

```diff
@@ -0,0 +1,32 @@
+<li class={{this.className}}>
+  <a
+    href={{this.url}}
+    title={{this.linkTitle}}
+    {{on "click" this.onClick}}
+  >
+    {{d-icon this.icon}}
+    <div>
+      {{#if this.label}}
+        {{#if this.wrapLabel}}
+          <span class={{this.labelWrapperClasses}}>
+            {{this.label}}
+          </span>
+        {{else}}
+          {{this.label}}
+        {{/if}}
+      {{/if}}
```

Comment:
I'm wondering if we can just always wrap the label in a `span` and avoid the extra conditional here. 

```suggestion
      {{#if this.label}}
        <span class={{this.labelWrapperClasses}}>
          {{this.label}}
        </span>
      {{/if}}
```

Reply:
I'm trying to preserve parity between the old widgets-based menu and this new menu with regard to the HTML structure. Some notification types only have a `label` and no `description` and their HTML looked like this in the old menu:

```html
<li>
  <a>
    <div>
      some label
    </div>
  </a>
</li>
```

If we render the label unconditionally in a span, then the HTML would look like this:

```html
<li>
  <a>
    <div>
      <span>some label</span>
    </div>
  </a>
</li>
```

I remember testing this and some notification types had broken styling. I considered updating the CSS, but I realized I'd have to make any changes in a backward compatible manner so that we don't break the styling of the old widgets-based menu. And there's also the question of themes/component that might expect certain HTML structure and break if we change the HTML, so I decided to just preserve parity and keep the HTML structure unchanged.

Reply:
:+1: Thanks for clarifying.